src/sidebar/store/modules/frames.ts

```diff
@@ -163,7 +163,16 @@ const mainFrame = createSelector(
  */
 const defaultContentFrame = createSelector(
   (state: State) => state.frames,
-  frames => firstFrameWithoutId(frames) ?? frames[0] ?? null,
+  frames => {
+    const mainFrame = firstFrameWithoutId(frames);
+    if (mainFrame) {
+      return mainFrame;
+    } else if (frames.length > 0) {
+      return frames[0];
+    } else {
+      return null;
+    }
+  },
```

Comment:
This now has the intended return type `Frame|null` instead of `Frame`. There is a TypeScript option to make indexing an array return `T|undefined` instead of `T`, but this is liable to cause a lot of spurious type errors in a large existing codebase. In modern browsers/Node, you can use `array.at(index)` in cases where you expect the index might be out of bounds, as this method has the return type `T|undefined`.